src/rofi_rbw/rofi_rbw.py

```diff
@@ -171,15 +185,21 @@ def get_data(self, name: str, folder: str) -> Data:
             encoding='utf-8'
         ).stdout.split('\n')
 
-        password = result[0].strip()
-        username = self.extract_username(result)
+        password = result[0]
+        username = self.extract_data("Username", result)
+        totp = self.extract_data("TOTP Secret", result)
+
+        if "pyotp" in sys.modules and totp != "":
+            totp = pyotp.parse_uri(totp).now()
+        else
+            totp = ""
 
-        return Data(username, password)
+        return Data(username, password, totp)
 
-    def extract_username(self, result: [str]) -> str:
+    def extract_data(self, key: str, result: [str]) -> str:
```

Comment:
I don't like that the caller now has to know how that `result` is structured. Previously, it called a method to extract the username, and didn't know or care how that worked. Now, it has to pass the `key`, which moves that knowledge.

But I also see that one wouldn't want to duplicate the code. So I see two ways of dealing with that: Either new, additional methods `extract_username` and `extract_totp` that call `extract_data`, or - and this is more involved but nicer, I think - an improved `credentials` class that is initialized with `rbw`s result (just the `str`) and offers the properties `password`, `username` and `totp`.
If you want, I could try my hand at the latter.

Reply:
Yea I see your point! I actually wrote a version that creates a `dict`, but that did lose multiple URI's and the notes. However considering only password/username/totp is needed I don't think it's a big issue. I personally also think that creating an `credentials` class which uses `rbw`s result is nicer. 

Rather than making a new class it's also possible to let `extract_data` return a `Data` object, however a separate class might give a better seperation of concerns.

Reply:
I created a new PR #12 to add the `Credentials` class and move the `extract_data` logic there. Maybe it's good to finish that one first, after which I'll rebase this PR and update the code accordingly.